asaps/models.py

```diff
@@ -0,0 +1,132 @@
+from asnake.client import ASnakeClient
+import importlib
+import json
+import attr
+
+
+@attr.s
+class Client:
+    authclient = attr.ib
+
+    def createclient(secfile):
+        """Select secrets.py file for the appropriate instance."""
+        secfileExists = importlib.util.find_spec(secfile)
+        if secfileExists is not None:
+            secrets = __import__(secfile)
+        else:
+            secrets = __import__('secrets')
+        print('Editing ' + secfile + ' ' + secrets.baseURL)
+        authclient = ASnakeClient(baseurl=secrets.baseURL,
+                                  username=secrets.user,
+                                  password=secrets.password)
+        authclient.authorize()
+        setattr(Client, 'authclient', authclient)
+
+    def getrecord(uri):
+        """Retrieve an individual record."""
+        record = Client.authclient.get(uri).json()
```

Comment:
Here you are treating the authclient as a class variable instead of an instance variable, which is not what we want. Change this line to:

```python
record = self.authclient.get(uri).json()
```